Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -309,6 +309,23 @@ PhysicalParticleContainer::AddGaussianBeam(Real x_m, Real y_m, Real z_m,
                   1, particle_w.dataPtr(),1);
 }
 
+void
+PhysicalParticleContainer::AddExternalFileBeam(std::string s_f)
+{
+#ifdef WARPX_USE_OPENPMD
+    openPMD::Series series = openPMD::Series(s_f, openPMD::AccessType::READ_ONLY);
+    amrex::Print() << "OpenPMD standard version " << series.openPMD() << "\n";
+    
+    openPMD::Iteration& i = series.iterations[1];
+    amrex::Print()  << "File contains " << i.particles.size() << " specie(s):";
+    for( auto const& ps : i.particles ) {
+        amrex::Print() << "\n\t" << ps.first;
```

Comment:
Newline after last species:
```suggestion
    amrex::Print()  << "File contains " << i.particles.size() << " specie(s):" << "\n";
    for( auto const& ps : i.particles ) {
        amrex::Print() << "\t" << ps.first << "\n";
```